libraries/cmake/source/glog/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2014-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+
+function(googleLogMain)
+  set(disabled_option_list
+    WITH_TLS
+    WITH_GFLAGS
+  )
+
+  set(enabled_option_list
+    HAVE_LIB_GFLAGS
+    WITH_THREADS
+  )
+
+  foreach(option_name ${disabled_option_list})
+    set("${option_name}" false CACHE BOOL "" FORCE)
+  endforeach()
+
+  foreach(option_name ${enabled_option_list})
+    set("${option_name}" true CACHE BOOL "" FORCE)
+  endforeach()
+
+  set(library_root "${CMAKE_CURRENT_SOURCE_DIR}/src")
+  add_subdirectory("${library_root}" "${CMAKE_CURRENT_BINARY_DIR}/submodule" EXCLUDE_FROM_ALL)
+  
+  set(compile_options_target_list
```

Comment:
Some of the projects we imported directly (add_subdirectory of the whole repo) are exporting targets; this means we can't directly target_link_libraries our interface targets like global_cxx_settings. @Smjert is improving the current implementation

Reply:
This is mostly done. I had to also add a system to patch submodule source code, because there's a nasty issue with googletest which is not solvable otherwise (save for using our own custom CMakeLists.txt).